### DISCUSSION TITLE
buf curl request failing but curl and grpcurl are successful... Why?

### DISCUSSION THREAD

**Discussion by abitofhelp** [connectrpc/connect-go#737] [Q&A] [answered]
Hello,

I am very familiar with gRPC and am experimenting with connectrpc for Go. I have a simple application based on the ubiquitous Greeter example at grpc's website. After generating, building, and running the server, curl and grpc requests are successful, but when I use "buf curl" the result is "not found". I'd appreciate any ideas how I can correct this error so I can experiment with Connect's protocol. I am using buf v1.32.0 and Go go1.22.3 darwin/amd64.

Thank you for your time and suggestions!

Here are the commands and results:

// GRPC
% grpcurl -plaintext -proto proto/abitofhelp/helloworld/v1/helloworld.proto -d '{"name": "Jane"}' localhost:8080 abitofhelp.helloworld.v1.GreeterService/SayHello
{
"message": "howdy Jane"
}

// HTTP
% curl -X POST -H "Content-Type: application/json" -d '{"name": "Jane"}' "http://127.0.0.1:8080/abitofhelp.helloworld.v1.GreeterService/SayHello"
{"message":"howdy Jane"}

// CONNECT
% buf curl --http2-prior-knowledge -d '{"name": "Jane"}' "http://127.0.0.1:8080/abitofhelp.helloworld.v1.GreeterService/SayHello"

Failure: failed to resolve symbol "abitofhelp.helloworld.v1.GreeterService": unimplemented: HTTP status 404 Not Found

**Comment by sudorandom** [comment, marked as answer]
@abitofhelp do you have server reflection enabled? I noticed that you gave `grpcurl` the knowledge of the protobuf file using the `-proto` option but you didn't do the same with `buf curl` using `--schema`. I suspect that 404 Not Found is from `buf curl` trying to reach the reflection API, which it will do whenever it isn't given a schema. You should be able to confirm this if you use the verbose flag.

So try this:
```
buf curl --http2-prior-knowledge --schema=proto -d '{"name": "Jane"}' "http://127.0.0.1:8080/abitofhelp.helloworld.v1.GreeterService/SayHello" -v
```

**Reply by abitofhelp** [reply]
Thank you for your help, @sudorandom!  You nailed it...  I missed the --schema command line option for buf. :)
